gpii/node_modules/flowManager/src/MatchMaking.js

```diff
@@ -92,13 +92,29 @@
      *      is fired with the modified mmpayload.
      */
     gpii.flowManager.getSolutions = function (solutionsRegistryDataSource, deviceContext, event, onError) {
+        gpii.flowManager.getSolutionsPromise(solutionsRegistryDataSource, deviceContext).then(
+            function (data) {
+                event.fire(data.solutionsRegistryEntries, data.solutions);
+            },
+            onError.fire
+        );
+    };
+
+    gpii.flowManager.getSolutionsPromise = function (solutionsRegistryDataSource, deviceContext) {
```

Comment:
Rather than adding a new `getSolutionsPromise` function, have you tried to get `getSolutions` return a promise besides firing events and have the capture component call `getSolutions`?

Reply:
Neat idea, I updated the function to handle both purposes, and cleaned up it's jsdoc